File: Ejemplos/Esp32/llm.c
#include <WiFi.h>
#include <HTTPClient.h>

const char* SSID = "TU_WIFI";
const char* PASS = "TU_PASS";
const char* URL  = "http://<PC_IP>:8000/llm";

void setup() {
  Serial.begin(115200);
  WiFi.begin(SSID, PASS);
  while (WiFi.status() != WL_CONNECTED) { delay(500); Serial.print("."); }
  Serial.println("\nWiFi OK");

  HTTPClient http;
  http.begin(URL);
  http.addHeader("Content-Type", "application/json");

  String payload = R"({
    "model":"qwen3-coder-30b",
    "temperature":0.2,
    "max_tokens":128,
    "top_p":0.9,
    "messages":[
      {"role":"system","content":"Sos un asistente útil."},
      {"role":"user","content":"Dame 2 ideas de proyectos con ESP32."}
    ]
  })";

  int code = http.POST(payload);
  Serial.println(code);
  String resp = http.getString();
  Serial.println(resp);
  http.end();
}

void loop(){}

File: system/Ejemplos/Python/llm.py
# pip install requests sseclient-py
import requests

URL = "http://localhost:8000/llm"

payload = {
    "model": "qwen3-coder-30b",
    "temperature": 0.2,
    "max_tokens": 256,
    "top_p": 0.9,
    "messages": [
        {"role": "system", "content": "Sos un asistente útil y conciso."},
        {"role": "user", "content": "Explicame qué es un debounce en electrónica."}
    ],
    # "stream": True  # si tu backend lo soporta y querés proxyear SSE del llama-server
}

r = requests.post(URL, json=payload, timeout=60)
r.raise_for_status()
print(r.json())
